### DISCUSSION TITLE
Retain file name when downloading recordings.

### DISCUSSION THREAD

**Discussion by antrose99** [foxglove/community#718] [Q&A]
Currently when you download a recording from the foxglove data platform the file will be named according to the Recording ID. Is it possible to make the file be named according to the original file name instead (or provide that as an option)? Currently I have to rename the files every time I download a recording as the ID does not really tell you anything.

**Comment by defunctzombie** [comment]
Thanks for the feedback! I've added it to our product ideas.

Internal ticket: FG-3560

**Comment by antrose99** [comment]
Downloading now seems to retain filename! Great!

Another peculiar thing I have found is that when downloading a recording via the Events page there is no name at all, and just downloads as "mcap". Is this expected? To me it would make sense to have some valid name, at the very least the sort of randomized filename. I have posted an image of what happens now but to replicate it should just be to try and download a recording from the events page.

![image](https://github.com/foxglove/community/assets/93656450/6d078848-bac0-40ec-b847-d5e87036c10e)

**Reply by wkalt** [reply]
Thanks - this is a bug and we're looking into it.

**Reply by wkalt** [reply]
@antrose99 this should be fixed now
